mlflow/utils/file_utils.py

```diff
@@ -357,6 +359,22 @@ def _copy_file_or_tree(src, dst, dst_dir=None):
     return dst_subpath
 
 
+def _copy_code_paths(code_paths, path, default_subpath="code"):
+    if code_paths is not None:
+        code_dir_subpath = default_subpath
+        for code_path in code_paths:
+            _copy_file_or_tree(src=code_path, dst=path, dst_dir=code_dir_subpath)
+    else:
+        code_dir_subpath = None
+    return code_dir_subpath
+
+
+def _validate_code_paths(code_paths):
+    if code_paths is not None:
+        if not isinstance(code_paths, list):
+            raise TypeError("Argument code_paths should be a list, not {}".format(type(code_paths)))
```

Comment:
Can we combine this validation logic with `copy_code_paths` to create a `validate_and_copy_code_paths` method? This way, flavor implementations don't need to import & call 2 separate methods. This is more robust, since it's somewhat easy for a developer to miss the validation method and still achieve working tests.